### ISSUE TITLE
Support for Keycloak SSO

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#497]
Keycloak is a popular, self-hosted OAuth provider (and more). This issue to track progress on adding it as a supported authentication mechanism in Firezone.

Current status: the ueberauth_keycloak_strategy Elixir library needs some fixes. Progress can be track on the Firezone fork here:

https://github.com/firezone/ueberauth_keycloak

**Comment by pschichtel** [comment]
Why focus on keycloak specifically instead of general openid connect and/or saml2?

**Comment by jamilbk** [comment]
@pschichtel Fair point.

We overhauled the auth system to [use a more modular authentication system](https://github.com/ueberauth/ueberauth/wiki/List-of-Strategies) for 0.3.0, and Google, Okta were the simplest to add first (and what most of our users were asking for).

I've begun work generalizing the OIDC auth system to expose more controls to support any OIDC provider and will focus on #501 to track work on that.

Closing this.
